lib/Compiler.js

```diff
@@ -64,29 +65,24 @@ function compileSolc(embark, contractFiles, cb) {
           compiled_object[className].functionHashes = contract.hashes;
           compiled_object[className].abiDefinition = JSON.parse(contract.abi);
           compiled_object[className].filename = fileName;
-
-          
         }
 
         fileCb();
       });
     },
     function (err) {
       cb(err, compiled_object);  
-      
-      embark.events.on("outputDone", function() {
-        if(outputBinary){
+      if(outputBinary){
+        embark.events.on("outputDone", function() {
           Object.keys(compiled_object).map(function(className, index) {
-            fs.writeFile(outputDir + className + ".bin", compiled_object[className].bin, (err) => {
+            fs.writeFile(path.join(outputDir, className + ".bin"), compiled_object[className].bin, (err) => {
```

Comment:
Can be `(path.join(outputDir, className, ".bin")`

Reply:
no, because it would add a "/"  before the extension .bin

Reply:
Ooh, my bad. I thought it was a directory (eg something/.bin/file). Forget my comment.